charts/k8s-service/templates/servicemonitor.yaml

```diff
@@ -0,0 +1,38 @@
+{{- if .Values.serviceMonitor.enabled }}
+---
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: {{ template "k8s-service.fullname" . }}
+  {{- if .Values.serviceMonitor.namespace }}
+  namespace: {{ .Values.serviceMonitor.namespace }}
+  {{- end }}
+  labels:
+    chart: {{ template "k8s-service.chart" . }}
+    app: {{ template "k8s-service.name" . }}
+    heritage: "{{ .Release.Service }}"
+    {{- if .Values.serviceMonitor.labels }}
+    {{- toYaml .Values.serviceMonitor.labels | nindent 4 }}
+    {{- end }}
+spec:
+  endpoints:
+    - interval: {{ .Values.serviceMonitor.interval }}
+      {{- if .Values.serviceMonitor.scrapeTimeout }}
+      scrapeTimeout: {{ .Values.serviceMonitor.scrapeTimeout }}
+      {{- end }}
+      {{- if .Values.serviceMonitor.honorLabels }}
+      honorLabels: {{ .Values.serviceMonitor.honorLabels }}
+      {{- end }}
+      {{- if .Values.serviceMonitor.port.name }}
+      port: {{ .Values.serviceMonitor.port.name }}
+      {{- end }}
+      {{- if .Values.serviceMonitor.path }}
+      path: {{ .Values.serviceMonitor.path }}
+      {{- end }}
+      {{- if .Values.serviceMonitor.scheme }}
+      scheme: {{ .Values.serviceMonitor.scheme }}
+      {{- end }}
```

Comment:
I wonder if this is better as an input that is directly rendered? E.g:

```
serviceMonitor:
  enabled: false
  endpoints:
    default:
      interval: 10s
      scrapeTimeout: 10s
      honorLabels: true
      port: http
      scheme: http
```

This way we can support:

- arbitrary number of endpoints in this list
- any additional config for the endpoints that are added in the future.

What do you think?